tests/test_fit_result.py

```diff
@@ -3,79 +3,106 @@
+        (dict(x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 3, b: 2}, 
+         {a: 0, b: 0}, LeastSquares, {'r_squared': 1.0, 'objective_value': 1e-23, 'chi_squared': 1e-23}),
+        # Test objective LeastSqueares
+        (dict(objective=LeastSquares, x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 3, b: 2}, 
+         {a: 1e-13, b: 1e-13}, LeastSquares, {'r_squared': 1.0, 'objective_value': 1e-23, 'chi_squared': 1e-23}),
+        # Test objective LogLikelihood
+        (dict(objective=LogLikelihood, x=np.linspace(1, 10, 10)), {a: 62.56756, b: 758.08369}, 
+         {a: float('nan'), b: float('nan')}, LogLikelihood, 
+         {'objective_value': -float('inf'), 'log_likelihood': float('inf'), 'likelihood': float('inf')})
+    ])
+def test_no_constraint(minimizer, fit_kwargs, expected_par, expected_std, expected_obj, expected_got):
+
+    # Values of a and b by COBYLA and TrustConstr of (objective = LeastSquares, None)
+    # Values of a and b always of (objective = LogLikeihood)
+    # LBFGSB has no status message/not string
+    fit = Fit(**fit_kwargs, minimizer=minimizer, model=Model({y: a * x ** b}))
     fit_result = fit.execute()
+    _run_tests(fit_result, expected_par, expected_std, expected_obj, expected_got)
 
-    assert isinstance(fit_result.params, OrderedDict)
 
+@pytest.mark.parametrize('minimizer',
+    subclasses(ConstrainedMinimizer) - {ChainedMinimizer, DifferentialEvolution} | {None}
+    )
+@pytest.mark.parametrize('fit_kwargs, expected_par, expected_std, expected_obj, expected_got',
+    [
+        # No special objective
+        (dict(x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 2.152889, b: 2.152889}, 
+         {a: 0.23715, b: 0.05076}, LeastSquares, {'r_squared': 0.99791, 'objective_value': 98.83587, 'chi_squared': 197.671746}),
+        # Test objective LeastSqueares
+        (dict(objective=LeastSquares, x=np.linspace(1, 10, 10), y=3 * np.linspace(1, 10, 10) ** 2), {a: 2.152889, b: 2.152889}, 
+         {a: 0.23715, b: 0.05076}, LeastSquares, {'r_squared': 0.99791, 'objective_value': 98.83587, 'chi_squared': 197.671746}),
+        # Test objective LogLikelihood
+        (dict(objective=LogLikelihood, x=np.linspace(1, 10, 10)), {a: 653.48460, b: 653.48460}, 
+         {a: float('nan'), b: float('nan')}, LogLikelihood, 
+         {'objective_value': -float('inf'), 'log_likelihood': float('inf'), 'likelihood': float('inf')})
+    ])
+def test_constraints(minimizer, fit_kwargs, expected_par, expected_std, expected_obj, expected_got):
+
+    # TrustConstr cannot handle LogLikelihood
+    # COBYLA cannot handle Eq
+    constraints = [
+        Eq(a, b),
+        CallableNumericalModel.as_constraint(
+            {z: ge_constraint}, connectivity_mapping={z: {a}},
+            constraint_type=Ge, model=Model({y: a * x ** b})
+        )
+    ]
+    fit = Fit(**fit_kwargs, minimizer=minimizer, model=Model({y: a * x ** b}), constraints=constraints)
```

Comment:
This call does not do what you want it to do if I look at the parameterizations and the accompanying comments. You specify the minimizer to use in the `fit_kwargs`. The minimizer passed is the *expected* minimizer, and should be passed to _run_tests. 
I'm surprised this doesnt cause errors

Reply:
I wanted to specify the minimizer of the Fit instance. I didn't mention a minimizer in `fit_kwargs`, just an objective :D `test_constraints` uses the same pattern as in `test_no_constraints`.

Reply:
True. My bad :)